scope_registry_test.go

```diff
@@ -221,3 +221,22 @@ func TestForEachScopeConcurrent(t *testing.T) {
 
 	<-done
 }
+
+func TestCachedReporterNoAllocOnOmitInternalMetrics(t *testing.T) {
+	r := newTestStatsReporter()
+	root, closer := NewRootScope(ScopeOptions{CachedReporter: r, OmitCardinalityMetrics: true}, 0)
```

Comment:
do you want to check that it gets allocated when `OmitCardinalityMetrics` is false?